### ISSUE TITLE
MD013/line-length enforced despite being disabled in config

### ISSUE THREAD

**Issue by cjustinobi** [DavidAnson/markdownlint-cli2#491]
I encountered an issue where the MD013/line-length rule is being enforced even though 
it is explicitly disabled in the configuration file. Here’s the relevant snippet from my config:

```bash
# MD013/line-length - Line length
# Changed from default so we can allow the paragraphs to be a single line
MD013: false

```

However, when I run the linter on my forked repository, the `MD013/line-length` rule is still checked, and errors are reported for lines exceeding the length limit.

### Steps to Reproduce

1. Use the following configuration file:

```bash
MD013: false
```

2. Run the linter on a file with long lines.

3. Observe that the `MD013/line-length` rule is enforced despite being disabled.

### Expected Behavior
The linter should not enforce the `MD013/line-length` rule when it is explicitly disabled in the configuration file.

### Actual Behavior
The linter enforces the `MD013/line-length` rule and reports errors for lines exceeding the length limit.

### Environment
Linter Version: DavidAnson/markdownlint-cli2-action@f5cf187ef11bd3a68a127321b794aa252ff23019
Node.js Version: v20

### Additional Context

I have verified the following:

 - The correct configuration file is being used.
 - No other configuration files are overriding this setting.

You can check this [PR](https://github.com/ethereum/portal-network-specs/pull/356) for more context.
